cmd/influxd/run.go

```diff
@@ -160,10 +159,34 @@ func (s *Node) closeSnapshotListener() error {
 	return err
 }
 
-func (cmd *RunCommand) Run(args ...string) error {
-	// Set up logger.
-	cmd.Logger = log.New(os.Stderr, "", log.LstdFlags)
+func (cmd *RunCommand) ParseConfig(path, hostname string) error {
+	var err error
+
+	// Parse configuration file from disk.
+	if path != "" {
+		cmd.config, err = ParseConfigFile(path)
+
+		if err != nil {
+			return fmt.Errorf("error parsing config %s - %s\n", path, err)
+		}
+		// Override config properties.
+		if hostname != "" {
+			cmd.config.Hostname = hostname
+		}
+
+		log.Printf("using config: %s\n", path)
```

Comment:
FWIW, I'd be a bit more explicit here since this is a log message for the user too.

`using configuration at ....`

whatever you think, I just like to be clearer.